pallets/ds-accounts/src/lib.rs

```diff
@@ -246,6 +284,28 @@ decl_module! {
             Ok(())
         }
 
+        #[weight = <T as Trait>::WeightInfo::register_uav()]
+        pub fn register_uav( origin, 
+                             uav_address: T::AccountId, 
+                             meta: T::MetaIPFS,
+                             serial_number: T::SerialNumber ) -> dispatch::DispatchResult {
+            let who = ensure_signed(origin)?;
+            let allow_registration = Self::account_is_registrar(&who) || Self::account_is_pilot(&who);
```

Comment:
inefficient check with double read. it's better to replace account_is_(x) with one  account_is(&account, roles)